applications/utils/reservation_creation.py

```diff
@@ -0,0 +1,63 @@
+import datetime
+import logging
+
+from django.db import Error
+
+from reservations.models import (
+    STATE_CHOICES,
+    RecurringReservation,
+    Reservation,
+    ReservationPurpose,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def create_reservations_from_allocation_results(application_event):
+    for schedule in application_event.application_event_schedules.all():
+        create_reservation_from_schedule_result(
+            schedule.applicationeventscheduleresult, application_event
+        )
+
+
+def create_reservation_from_schedule_result(result, application_event):
+    recurring_reservation = RecurringReservation.objects.create(
+        user=application_event.application.user,
+        application=application_event.application,
+        age_group=result.application_event_schedule.application_event.age_group,
+        ability_group=result.application_event_schedule.application_event.ability_group,
+    )
+
+    reservation_date = application_event.begin
+    if not reservation_date.weekday() == result.allocated_day:
+        reservation_date = reservation_date + datetime.timedelta(
+            days=abs(reservation_date.weekday() - result.allocated_day)
+        )
+    interval = 14 if application_event.biweekly else 7
+    while reservation_date < application_event.end:
+        try:
+            reservation = Reservation.objects.create(
```

Comment:
There probably should be a check for overlaps

There is a ReservationUnit#check_reservation_overlap

which could be used.

Reply:
added overlap logic to create the reservation as DENIED status if overlaps.

Reply:
👍 denied sounds better than cancelled that i suggested in the meeting.